### PR TITLE
envelope: init at 0.0.4

### PR DESCRIPTION
**Pull request by xiorcale**
###### Motivation for this change

Add more curated apps from Elementary OS app store for Pantheon Desktop.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by mmahut** [comment]
@GrahamcOfBorg build envelope

**Comment by mmahut** [comment]
@GrahamcOfBorg build envelope

File: pkgs/applications/office/envelope/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    owner = "cjfloss";
+    repo = pname;
+    rev = version;
+    sha256 = "111lq1gijcm7qwpac09q11ymwiw2x3m12a28ki52f28fb1amvffc";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pantheon.vala
```
Comment (worldofpeace) [inline pkgs/applications/office/envelope/default.nix]: this attribute is gone, as I've mentioned.

File: pkgs/applications/office/envelope/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  version = "0.0.4";
+
+  src = fetchFromGitHub {
+    owner = "cjfloss";
+    repo = pname;
+    rev = version;
+    sha256 = "111lq1gijcm7qwpac09q11ymwiw2x3m12a28ki52f28fb1amvffc";
+  };
+
+  nativeBuildInputs = [
```
Comment (worldofpeace) [inline pkgs/applications/office/envelope/default.nix]: please add `gettext`

File: pkgs/applications/office/envelope/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    meson
+    ninja
+    pantheon.vala
+    pkgconfig
+    python3
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    git
```
Comment (worldofpeace) [inline pkgs/applications/office/envelope/default.nix]: git here appears to be optional https://github.com/cjfloss/envelope/blob/master/src/meson.build#L8 and we shouldn't need it.

File: pkgs/applications/office/envelope/default.nix [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    appstream-glib
```
Comment (worldofpeace) [inline pkgs/applications/office/envelope/default.nix]: can you move this to `nativeBuildInputs` and also add `desktop-file-utils`?
* https://github.com/cjfloss/envelope/blob/master/data/meson.build#L33

On another note, I should probably package their vala linting tool 
* https://github.com/cjfloss/envelope/blob/master/data/meson.build#L51

**Review by worldofpeace — COMMENTED** [review]
Can you add `doCheck = true;` so we can run the test?
* https://github.com/cjfloss/envelope/blob/master/tests/meson.build#L33

File: pkgs/applications/office/envelope/default.nix (hunk 1 above)
Comment (Kjuvi) [reply pkgs/applications/office/envelope/default.nix]: > Can you add `doCheck = true;` so we can run the test?
> 
>     * https://github.com/cjfloss/envelope/blob/master/tests/meson.build#L33

Interestingly the test failed to validate the appdata file. I updated the PR with the failed test but I have no clue about what I should do to correct that ?

**Comment by Kjuvi** [comment]
> Can you add `doCheck = true;` so we can run the test?
> 
>     * https://github.com/cjfloss/envelope/blob/master/tests/meson.build#L33

Interestingly the test failed to validate the appdata file. I updated the PR with the failed test but I have no clue about what I should do to correct it.

File: pkgs/applications/office/envelope/default.nix (hunk 4 above)
Comment (Kjuvi) [reply pkgs/applications/office/envelope/default.nix]: I saw the missing vala linting tool too. Could be great if we have it. Packaging it don't seem too difficult, so if I find the time I can maybe try to submit a PR too.

Comment (worldofpeace) [reply pkgs/applications/office/envelope/default.nix]: You're in luck, I did this yesterday https://github.com/NixOS/nixpkgs/pull/69052/commits/c7cc5ab07bac39acf673d22c85ae21296b3ac2a2

**Comment by worldofpeace** [comment]
> > Can you add `doCheck = true;` so we can run the test?
> > ```
> > * https://github.com/cjfloss/envelope/blob/master/tests/meson.build#L33
> > ```
> 
> Interestingly the test failed to validate the appdata file. I updated the PR with the failed test but I have no clue about what I should do to correct it.

So probably our appstream-glib is out of date, or their appstream file needs corrections.
Will look into this.

**Comment by worldofpeace** [comment]
@Kjuvi Opened https://github.com/cjfloss/envelope/pull/59 for the issues with appdata validation.
It was set to a strict validation, and it wanted network access which is disallowed in the nix sandbox.

**Pull request merged**